server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -379,6 +379,12 @@ private void snapshot(
             if (indexShard.routingEntry().primary() == false) {
                 throw new IndexShardSnapshotFailedException(shardId, "snapshot should be performed only on primary");
             }
+            if (indexShard.indexSettings().isSegRepEnabled() && indexShard.isPrimaryMode() == false) {
```

Comment:
1. Can we add a unit test for this change ?
2. Do we need to handle this for `DOCUMENT` replication enabled indices ?

Reply:
I tried for a while here to add a unit level test but it's not straight forward. The indexshard level tests do not go through the service class and validation here on service class itself is layers deep in private methods. 

As for docrep I would rather not unless there is a case for it?

Reply:
added a unit test.